most.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const most = require('most');
+const R = require('ramda');
+const S = require('sanctuary');
+
+
+//    join :: String -> String -> String
+const join = R.curryN(2, path.join);
+
+//    readFile :: String -> String -> Promise Error String
+const readFile = R.curry((encoding, filename) =>
+  new Promise((resolve, reject) =>
+    fs.readFile(filename, { encoding }, (err, data) =>
+      err != null ? reject(err) : resolve(data)
+    )));
+
+//    concatFiles :: String -> Promise Error String
+const concatFiles = dir =>
+  R.pipe(R.pipe(readFile('utf8'), most.fromPromise),
+         R.map(S.lines),
+         R.chain(R.pipe(R.map(join(dir)), most.from)),
+         R.map(readFile('utf8')),
+         stream => stream.await(),
+         R.reduce(R.concat, ''))(join(dir, 'index.txt'));
```

Comment:
At first glance `R.reduce(R.concat, ''))(join(dir, 'index.txt'))` looks like a single expression.

Let's use this formatting:

```javascript
R.pipe(
  ...,
  ...,
  ...,
  ...,
  ...,
  ...
)(join(dir, 'index.txt'))
```